Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -1155,6 +1158,10 @@ size_t
 GroupDetectors2::formGroupsEvent(DataObjects::EventWorkspace_const_sptr inputWS,
                                  DataObjects::EventWorkspace_sptr outputWS,
                                  const double prog4Copy) {
+  if (inputWS->detectorInfo().isScanning())
+    throw std::runtime_error("GroupDetectors does not currently support "
+                             "workspaces with detector scans.");
```

Comment:
Could you add a unit test for this?

Also, maybe explicitly mention `EventWorkspace` in the error message?